src/main/java/games/strategy/engine/chat/Chat.java

```diff
@@ -55,6 +55,57 @@
     LOBBY_CHATROOM, GAME_CHATROOM, NO_SOUND
   }
 
+  public Chat(final IMessenger messenger, final String chatName, final IChannelMessenger channelMessenger,
+      final IRemoteMessenger remoteMessenger, final CHAT_SOUND_PROFILE chatSoundProfile) {
+
+    this.chatSoundProfile = chatSoundProfile;
+    this.messengers = new Messengers(messenger, remoteMessenger, channelMessenger);
+    statusManager = new StatusManager(messengers);
+    chatChannelName = ChatController.getChatChannelName(chatName);
+    this.chatName = chatName;
+    sentMessages = new SentMessagesHistory();
+
+    // the order of events is significant.
+    // 1 register our channel listener
+    // once the channel is registered, we are guaranteed that
+    // when we receive the response from our init(...) message, our channel
+    // subscriber has been added, and will see any messages broadcasted by the server
+    // 2 call the init message on the server remote. Any add or join messages sent from the server
+    // will queue until we receive the init return value (they queue since they see the init version is -1)
+    // 3 when we receive the init message response, acquire the lock, and initialize our state
+    // and run any queued messages. Queued messages may be ignored if the
+    // server version is incorrect.
+    // this all seems a lot more involved than it needs to be.
+    final IChatController controller = (IChatController) messengers.getRemoteMessenger()
+        .getRemote(ChatController.getChatControlerRemoteName(chatName));
+    messengers.getChannelMessenger().registerChannelSubscriber(m_chatChannelSubscribor,
+        new RemoteName(chatChannelName, IChatChannel.class));
+    final Tuple<Map<INode, Tag>, Long> init = controller.joinChat();
+    final Map<INode, Tag> chatters = init.getFirst();
+    synchronized (mutexNodes) {
+      nodes = new ArrayList<>(chatters.keySet());
+    }
+    chatInitVersion = init.getSecond();
+    synchronized (mutexQueue) {
+      queuedInitMessages.forEach(Runnable::run);
+      assignNodeTags(chatters);
+      queuedInitMessages = null;
+    }
+    updateConnections();
+  }
+
+  private void updateConnections() {
+    synchronized (mutexNodes) {
+      if (nodes == null) {
+        return;
+      }
+      final List<INode> playerNames = new ArrayList<>(nodes);
+      Collections.sort(playerNames);
+      for (final IChatListener listener : listeners) {
+        listener.updatePlayerList(playerNames);
+      }
+    }
+  }
   private void addToNotesMap(final INode node, final Tag tag) {
```

Comment:
Here's the Checkstyle violation: missing a newline between the end of the previous method and the start of this one.

Reply:
Thanks! fixed in 1e18d8e